yourgov/components/FormMobileFoodVendorPermit/FormTask1FormState.ts

```diff
@@ -0,0 +1,156 @@
+import { DeepPartial } from 'react-hook-form';
+import * as yup from 'yup';
+import { yupDateField, yupPhoneField } from './utils';
+
+export const task1Step1FormSchema = yup
+	.object({
+		firstName: yup.string().required('Enter your first name'),
+		lastName: yup.string().required('Enter your last name'),
+		email: yup
+			.string()
+			.email('Enter a valid email')
+			.required('Enter your email'),
+		contactPhoneNumber: yupPhoneField.optional(),
+	})
+	.required();
+
+export type Task1Step1FormSchema = yup.InferType<typeof task1Step1FormSchema>;
+
+export const task1Step1Part2FormSchema = yup
+	.object({
+		contactPhoneNumber: yupPhoneField.optional(),
+	})
+	.required();
+
+export type Task1Step1Part2FormSchema = yup.InferType<
+	typeof task1Step1Part2FormSchema
+>;
+
+export const task1Step2FormSchema = yup
+	.object({
+		businessName: yup.string().required('Business or company name is required'),
+		tradingName: yup.string(),
+		businessStructure: yup
+			.string()
+			.typeError('Business structure is required')
+			.required('Business structure is required'),
+		abn: yup.string().when('businessStructure', (value, schema) => {
+			if (value === 'Business') {
+				return schema.required('ABN is required');
+			}
+			return schema;
+		}),
+	})
+	.required();
+
+export type Task1Step2FormSchema = yup.InferType<typeof task1Step2FormSchema>;
+
+export const task1Step3FormSchema = yup
+	.object({
+		// street address
+		streetAddress: yup.string().required('Enter your street address'),
+		suburbTownCity: yup.string().required('Enter your suburb, town or city'),
+		state: yup.string().required('Enter your state'),
+		postcode: yup.string().required('Enter your postcode'),
+		// postal address
+		isPostalAddressSameAsStreetAddress: yup.boolean(),
+		postalAddress: yup.string().when('isPostalAddressSameAsStreetAddress', {
+			is: false,
+			then: yup.string().required('Enter your postal address'),
+		}),
+		postalSuburbTownCity: yup
+			.string()
+			.when('isPostalAddressSameAsStreetAddress', {
+				is: false,
+				then: yup.string().required('Enter your suburb, town or city'),
+			}),
+		postalState: yup.string().when('isPostalAddressSameAsStreetAddress', {
+			is: false,
+			then: yup.string().required('Enter your state'),
+		}),
+		postalPostcode: yup.string().when('isPostalAddressSameAsStreetAddress', {
+			is: false,
+			then: yup.string().required('Enter your postcode'),
+		}),
+	})
+	.required();
+
+export type Task1Step3FormSchema = yup.InferType<typeof task1Step3FormSchema>;
+
+export const task1Step4FormSchema = yup
+	.object({
+		registrationNumber: yup
+			.string()
+			.max(6, 'Register number can not be longer than 6 characters')
+			.required('Vehicle registration number is required'),
```

Comment:
I don't know if this is a problem with our patterns or this implementation, but error corrections can be incredibly jarring.

Steps to reproduce:

1. Submit empty vehicle registration form
2. Type a character in rego field
3. Delete the character
4. Repeat ad inifinitum
5. The error Page Alert keeps disappearing and re-appearing

Reply:
[The pattern can be found here](https://design-system.agriculture.gov.au/patterns/accessible-form-validation-and-recovery#invalid-form-fields] which says to only show the page alert when there is 1 error.

Since a single character would technically be a valid registration number, it makes sense why the page alert is disappearing and re-appearing. But agree this is jarring.

Reply:
To discuss later outside of this PR.